x-pack/plugins/lens/public/xy_visualization/axis_settings_popover.tsx

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiSwitch,
+  EuiSpacer,
+  EuiFieldText,
+  IconType,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { LayerConfig, AxesSettingsConfig } from './types';
+import { FramePublicAPI } from '../types';
+import { ToolbarPopover } from '../shared_components';
+import { ToolbarButtonProps } from '../toolbar_button';
+// @ts-ignore
```

Comment:
My dear @cchaos can we somehow avoid ts-ignore here?

Reply:
It should be doable like this:
* Rename `axis_*.js` -> `axis_*.ts`
* Change `({ title, titleId, ...props })` -> `({ title, titleId, ...props }: : { title: string; titleId: string })`